src/app/script/sprite_class.cpp

```diff
@@ -909,6 +909,20 @@ int Sprite_set_filename(lua_State* L)
   return 0;
 }
 
+int Sprite_set_isModified(lua_State* L)
+{
+  auto sprite = get_docobj<Sprite>(L, 1);
+  const bool isModified = lua_toboolean(L, 2);
+  Doc* doc = static_cast<Doc*>(sprite->document());
+  if (doc && !isModified)
+    doc->markAsSaved();
```

Comment:
This should be done by the FOP code instead of leaving this to the script, e.g. if the `onsave` function doesn't fail, it means that the file was saved successfully, but if we have just exported the file to this plugin file format, we don't have to mark the sprite as saved. For example see

https://github.com/aseprite/aseprite/blob/3c350c3e671fe5755a9ca098a13e3bd26b86bf5b/src/app/commands/cmd_save_file.cpp#L249-L253

The sprite is marked as saved only when "markAsSaved" flag is `MarkAsSaved::On` (which is not the case for `SaveFileCopyAsCommand`).